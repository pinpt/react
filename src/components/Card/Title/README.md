### The title for the content card

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-card-title) | [Github](https://github.com/pinpt/react/tree/master/src/components/Card/Title)

#### Theming

The parent className for styling this component is `.Pinpoint.Content.Card.Title`

<details>
	<summary>Base Theme Styles</summary>

```css
.Pinpoint.Content.Card.Title {
	@apply text-lg font-semibold transition-colors duration-200;
	color: var(--card-title-color);
}

.Pinpoint.Content.Card:hover .Title {
	color: var(--card-link-color);
}
```

</details>
