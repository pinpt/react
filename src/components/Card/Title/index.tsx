import React from 'react';
export interface ICardTitleProps {
	className?: string;
	title: string;
}

const Title = (props: ICardTitleProps) => {
	const { className, title } = props;
	return <h2 className={`Pinpoint Content Card Title ${className ?? ''}`}>{title}</h2>;
};

export default Title;
