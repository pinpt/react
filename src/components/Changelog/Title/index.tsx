export interface IChangelogTitleProps {
	className?: string;
	title: string;
}

const ChangelogTitle = (props: IChangelogTitleProps) => {
	const { className, title } = props;
	return (
		<h2 className={`Changelog Title ${className ?? ''}`}>
			{title}
		</h2>
	);
};

export default ChangelogTitle;
