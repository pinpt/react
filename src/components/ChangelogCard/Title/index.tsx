export interface IChangelogTitleProps {
	className?: string;
	title: string;
}

const ChangelogTitle = (props: IChangelogTitleProps) => {
	const { className, title } = props;
	return (
		<h2 className={`Changelog Card Title ${className ?? ''}`}>
			{title}
		</h2>
	);
};

export default ChangelogTitle;
