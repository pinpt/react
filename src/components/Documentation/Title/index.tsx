import Logo from '../../Logo';
import { ISite } from '../../../lib';

export interface IDocoumentationTitleProps {
	className?: string;
	site?: ISite;
	text?: string;
	onClick?: () => void;
}

const Title = (props: IDocoumentationTitleProps) => {
	const { className = '', site, text, onClick } = props;
	return (
		<div className={`Pinpoint Documentation Title ${onClick ? 'clickable' : ''} ${className}`} onClick={onClick}>
			{site && <Logo src={site.logoUrl} />}
			<div className="text">{text ? text : site?.name}</div>
		</div>
	);
};

export default Title;
