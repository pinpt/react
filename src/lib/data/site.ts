import type { Site } from '../types/site';
import { executeAPI } from '../fetch';

export const fetchSite = async (siteId: string): Promise<Site> => {
	const { site } = await executeAPI(`/site-api/v1/site/${siteId}`);
	return site;
};

type AnalyticsResult = Record<string, { claps: number; pageviews: number }>;

export const fetchAnalytics = async (siteId: string): Promise<AnalyticsResult> => {
	const { data } = await executeAPI(`/site-api/v1/site/${siteId}/analytics`);
	return data;
};
