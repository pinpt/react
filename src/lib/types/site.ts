export interface ISite {
	id: string;
	logoUrl?: string;
	mailingAddress?: string;
	url: string; // the canonical url for the site
	slug: string;
	name: string;
	private: boolean; // true if private and should not be indexable by search engines
	theme?: {
		title?: string;
		social?: Record<string, string>;
		description?: string;
		logoLink?: string;
		copyright?: string;
		eulaLink?: string;
		privacyLink?: string;
		tosLink?: string;
		homepage?: {
			tags?: string[];
		};
	};
	scriptPlugins?: Record<string, any>;
	basePath: string;
}
