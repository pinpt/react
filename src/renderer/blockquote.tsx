import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
