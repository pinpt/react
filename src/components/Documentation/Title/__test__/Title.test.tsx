import renderer from 'react-test-renderer';
import Title from '..';
import site from '../../../Prebuilt/__data__/testSite.json';

test('Test default state', () => {
	const component = renderer.create(<Title site={site} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom text', () => {
	const component = renderer.create(<Title site={site} text="Test Cutom Title Text" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom className', () => {
	const component = renderer.create(<Title site={site} className="test-custom" />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test click handler', () => {
	const component = renderer.create(<Title site={site} onClick={() => console.log('click!')} />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
