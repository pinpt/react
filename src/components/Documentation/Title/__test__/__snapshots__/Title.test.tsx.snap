// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom className 1`] = `
<div
  className="Pinpoint Documentation Title test-custom"
>
  <span
    className="Pinpoint Internal Link  Pinpoint Logo sm "
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <div
    className="text"
  >
    Pinpoint
  </div>
</div>
`;

exports[`Test custom text 1`] = `
<div
  className="Pinpoint Documentation Title "
>
  <span
    className="Pinpoint Internal Link  Pinpoint Logo sm "
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <div
    className="text"
  >
    Test Cutom Title Text
  </div>
</div>
`;

exports[`Test default state 1`] = `
<div
  className="Pinpoint Documentation Title "
>
  <span
    className="Pinpoint Internal Link  Pinpoint Logo sm "
  >
    <img
      className="image"
      src="https://file.pinpoint.com/1fcde4196a4c70a8a86f0ce4af53f2a5;U75%3B-sj_NAoPtDojoij_RdfPRdaxoPfPavaw;320x320.png"
    />
  </span>
  <div
    className="text"
  >
    Pinpoint
  </div>
</div>
`;
